base/experimental.jl

```diff
@@ -494,4 +494,78 @@ function entrypoint(@nospecialize(argt::Type))
     nothing
 end
 
+### Task metrics
+
+"""
+    Base.Experimental.task_metrics(::Bool)
+
+Enable or disable the collection of per-task metrics.
+A `Task` created when `Base.Experimental.task_metrics(true)` is in effect will have
+[`Base.Experimental.task_cpu_time_ns`](@ref) and [`Base.Experimental.task_wall_time_ns`](@ref)
+timing information available.
+
+!!! note
+    Task metrics can be enabled at start-up via the `--task-metrics=yes` command line option.
+"""
+function task_metrics(b::Bool)
+    if b
+        ccall(:jl_task_metrics_enable, Cvoid, ())
+    else
+        ccall(:jl_task_metrics_disable, Cvoid, ())
+    end
+    return nothing
 end
+
+"""
+    Base.Experimental.task_cpu_time_ns(t::Task) -> Union{UInt64, Nothing}
+
+Return the total nanoseconds that the task `t` has spent running.
+This metric is only updated when `t` yields or completes unless `t` is the current task, in
+which it will be updated continuously.
+See also [`Base.Experimental.task_wall_time_ns`](@ref).
+
+Returns `nothing` if task timings are not enabled.
+See [`Base.Experimental.task_metrics`](@ref).
+
+!!! note "This metric is from the Julia scheduler"
+    A task may be running on an OS thread that is descheduled by the OS
+    scheduler, this time still counts towards the metric.
+
+!!! compat "Julia 1.12"
+    This method was added in Julia 1.12.
+"""
+function task_cpu_time_ns(t::Task=current_task())
```

Comment:
This name seems like it might be a bit mis-leading - What about `task_running_time_ns()`?

As the note clarifies, this is the time that the task has been "in the running state" (i.e. scheduled by Julia) but not the time the task has actually spent running on-cpu (i.e. scheduled by the OS)

It's plausible to me that one day we might want to hook up `perf` with our scheduler so that you can get the actual instruction / branch / tsc spent on-core, so it's probably good to leave room for the future names

Reply:
That seems like good feedback to me. 👍 
I'm supportive of a name change.

I like `task_running_time_ns()`. I also like `task_executing_time_ns()` or maybe `task_scheduled_time_ns()`?
I think `task_running_time_ns` is the best I've seen so far.

Reply:
Changed.